account_invoice_export/models/account_move.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import requests
+
+import odoo
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+from odoo.addons.queue_job.job import job
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    invoice_exported = fields.Boolean(copy=False)
+    # Usefull when the distant system does not validate the export synchronously
+    invoice_export_confirmed = fields.Boolean(copy=False)
+
+    def export_invoice(self):
+        for invoice in self:
+            description = "{} - Export ebill".format(invoice.transmit_method_id.name)
+            invoice.with_delay(description=description)._job_export_invoice()
+
+    @job(default_channel="root.invoice_export")
+    def _job_export_invoice(self):
+        """Export ebill to external server and update the chatter."""
+        self.ensure_one()
+        if self.invoice_exported and self.invoice_export_confirmed:
+            return _("Nothing done, invoice has already been exported before.")
+        try:
+            res = self._export_invoice()
+        except Exception as e:
+            values = {
+                "job_id": self.env.context.get("job_uuid"),
+                "error_detail": "",
+                "error_type": type(e).__name__,
+                "transmit_method_name": self.transmit_method_id.name,
+            }
+            if type(e) == UserError:
```

Comment:
To catch other types of Odoo-side errors, you could do
```suggestion
            if isinstance(e, except_orm):
```
`except_orm` is the base class of `UserError` and some others.

Reply:
Changed